sandbox/DemoApp.qml

```diff
@@ -564,12 +564,150 @@ Rectangle {
+                        subMenuItemIcons: [
+                            "chat",
+                            {
+                                source: "https://pbs.twimg.com/profile_images/1369221718338895873/T_5fny6o_400x400.jpg"
+                            },
+                            { isLetterIdenticon: true, color: "red" }
+                        ]
+
+                        StatusMenuItem {
+                            text: "Anywhere"
+                            onTriggered: {
+                                searchPopup.resetSelectionBadge();
+                                searchPopup.selectionBadgePrimaryText = text;
+                            }
+                        }
+
+                        StatusMenuSeparator {}
+
+                        StatusPopupMenu {
+                            title: "Chat"
+
+                            StatusMenuItem {
+                                text: "vitalik.eth"
+                                image.source: "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAyCAYAAAAeP4ixAAAAlklEQVR4nOzW0QmDQBAG4SSkl7SUQlJGCrElq9F3QdjjVhh/5nv3cFhY9vUIYQiNITSG0Bh
+                                    CExPynn1gWf9bx498P7/nzPcxEzGExhBdJGYihtAYQlO+tUZvqrPbqeudo5iJGEJjCE15a3VtodH3q2ImYgiNITTlTdG1nUZ5a92VITQxITFiJmIIjSE0htAYQrMHAAD//+wwFVpz+yqXAAAAAElFTkSuQmCC"
+                                image.isIdenticon: true
+                                onTriggered: {
+                                    searchPopup.resetSelectionBadge();
+                                    searchPopup.selectionBadgePrimaryText = text;
+                                    searchPopup.selectionBadgeImage = image.source;
+                                }
+                            }
+
+                            StatusMenuItem {
+                                text: "Pascal"
+                                image.source: "https://pbs.twimg.com/profile_images/1369221718338895873/T_5fny6o_400x400.jpg"
+                                onTriggered: {
+                                    searchPopup.resetSelectionBadge();
+                                    searchPopup.selectionBadgeImage = image.source;
+                                    searchPopup.selectionBadgePrimaryText = text;
+                                }
+                            }
+                        }
+
+                        StatusPopupMenu {
+                            id: cryptoKittiesPopupMenu
+                            title: "Cryptokitties"
+
+                            StatusMenuItem {
+                                text: "welcome"
+                                iconSettings.name: "channel"
+                                iconSettings.color: Theme.palette.directColor1
+                                onTriggered: {
+                                    searchPopup.resetSelectionBadge();
+                                    searchPopup.selectionBadgeImage = cryptoKittiesPopupMenu.currentImg;
```

Comment:
There is a problem with StatusPopupMenu, you can try printing `currentOptionItemImgSource` for both 'Chat' and 'Cryptokitties' in DemoApp.qml  (on `Component.onCompleted`) and you'll get 
```
qml: Cryptokitties 
qml: Chat https://pbs.twimg.com/profile_images/1369221718338895873/T_5fny6o_400x400.jpg
```
so `currentOptionItemImgSource` for 'Cryptokitties' is always empty.

Reply:
So... the url shown in `Chat` however is the image that belongs to `CryptoKitties` in this case. Probably a off-by-one bug?

Reply:
Not sure why, however as this is relevant to StatusPopupMenu and not the SearchPopup, should we look fix this in this task or maybe file another bug?

Reply:
The currentOptionItemImgSource and other properties have been introduced in this pr so we should also fix them here, otherwise we introduce this bug in the First place just to fix it later.